### ISSUE TITLE
Investigate removal of internal self-attach APIs used by localConnector-1.0

### ISSUE THREAD

**Issue by aguibert** [OpenLiberty/open-liberty#6989]
Currently the `localConnector-1.0` feature allows for JVM self-attach, which is often used by tooling.

The feature, specifically `com.ibm.ws.jmx.connector.local.LocalConnectorActivator`, depends on 2 internal JDK APIs:
- `jdk.internal.agent.Agent`
- `jdk.internal.vm.VMSupport`

We should investigate if there is a different, more proper way to do self-attach, since these internal APIs are A) moved from JRE to JDK in Java 11 and B) subject to change/removal without notice in a future version of Java.

**Comment by aguibert** [comment]
The problem code is in `LocalConnectorActivator`, which does 3 things:
1. Invokes `jdk.internal.agent.Agent.agentmain(null)` to start the local JMX agent.
2. Read the local connector address from the agent properties
3. Write the local connector address to a file, which can be read by other processes (such as WDT or Arquillian)

The code to start the JMX agent lives in the `jdk.management.agent` module. So I think requiring this module in order to use the `localConnector-1.0` feature is a must. This module is not present in a JRE.

Looking past the JRE limitation, when the `jdk.management.agent` module _is_ present, we still have an issue because to perform step (1) and (2) we have to bust into the `jdk.internal.agent` package to invoke `Agent.agentmain()`. According to JDK source, there are 2 proper ways to start the agent:
```
1. startAgent() method is invoked by the VM if -Dcom.sun.management.* is set
2. startLocalManagementAgent() or startRemoteManagementAgent()
   is invoked to start the management agent after the VM starts
   via jcmd ManagementAgent.start and start_local command.
```

The `jconsole` utility is able to get around this restriction because the `jdk.management.agent` module explicitly grants access to its internal APIs from the `jdk.jconsole` module:
```java
module jdk.management.agent {
    requires java.management;
    requires java.management.rmi;

    exports jdk.internal.agent to jdk.jconsole;

    uses jdk.internal.agent.spi.AgentProvider;
}
```

Also, looking into JDK source, JConsole performs steps (1) and (2) exactly as we do in `LocalConnectorActivator`, using the same `jdk.internal.*` APIs.

Based on this info, we cannot improve upon our internal API usage in the `localConnector-1.0` feature.

However, all of my investigation has revolved around starting a JMX agent and making its address available. Perhaps someone who understands more about how we use the JMX agent can suggest an alternative path. For example, perhaps WDT can start using a REST API instead of a JMX agent?

**Comment by aguibert** [comment]
Discussed with Alasdair and he suggested we do:
```java
      VirtualMachine vm = VirtualMachine.attach(String.valueOf(ProcessHandle.current().pid()));
      System.out.println(vm.startLocalManagementAgent());
```
which only works if the `-Djdk.attach.allowAttachSelf=true` system property is set, which we can start setting in the server start script all the time.

**Comment by aguibert** [comment]
All of the code for this story has been delivered and should be included in 19.0.0.8.
